### DISCUSSION TITLE
Is it possible to provide custom `vscode.TreeItem`

### DISCUSSION THREAD

**Discussion by prashantvc** [microsoft/vscode-discussions#555] [Extension Development QnA] [answered]
I want to show `TreeItem` similar to the extension view, a large Icon, title and the description. 
Is it possible to extend `vscode.TreeItem` to look like, for example like this

<img width="409" alt="image" src="https://user-images.githubusercontent.com/328076/229551351-6b937ff3-cac7-41ed-96c7-4bd65db63e06.png">

**Comment by alefragnani** [comment, marked as answer]
Hi @prashantvc ,

Unfortunately, this is not possible today. To achieve such UI you should move to WebView approach (https://code.visualstudio.com/api/ux-guidelines/webviews).

I remember some questions about it a while ago, but didn't find references in VS Code repo, so I think this may have happened in SO or Slack channel. I didn't find any open issue about it as well.

Hope this helps

**Comment by sketchbuch** [comment]
With a webview you can. My extension (https://github.com/sketchbuch/vsc-workspace-sidebar) used to be a treeview but I converted it to a webview to be able to do more than tree view offered, maybe looking at the code will help you if you get stuck.
